lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1294,12 +1302,12 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
     return processCommandLineAndRunImmediately(
         Invocation, Instance, std::move(SM), MSF, observer, ReturnValue);
 
+  const std::string OutputFilename = PSPs.OutputFilename;
```

Comment:
No need to eagerly copy this, right? (StringRef)

Reply:
If, at some time in the future, Invocation is discarded before code generation is finished, using a StringRef would introduce a bug. Do you want to depend on the Invocation not being freed?

Reply:
I don't think that could happen, since `generateIR` doesn't get a reference to the Invocation. (Also, PSPs gets passed to `generateIR` right now anyway.)

Reply:
Thanks for explaining. I've fixed this and also parallelOutputFilenames to avoid copying as they go down the stack.